Vic2ToHoI4/Source/HOI4World/HoI4FocusTree.cpp

```diff
@@ -1906,65 +1989,589 @@ void HoI4FocusTree::addGPWarBranch(shared_ptr<HoI4::Country> Home, const vector<
+			}
+
+			if (const auto& originalFocus = loadedFocuses.find("declare_war"); originalFocus != loadedFocuses.end())
+			{
+				shared_ptr<HoI4Focus> newFocus = originalFocus->second.makeTargetedCopy(theCountry->getTag(), country.first);
+				newFocus->prerequisites.clear();
+				newFocus->prerequisites.push_back("= { focus = war_plan" + theCountry->getTag() + country.first + " }");
+				newFocus->relativePositionId += country.first;
+				if (!majorIdeologies.count("fascism"))
+				{
+					newFocus->updateFocusElement(newFocus->aiWillDo, fascistGovernmentCheck, "");
+				}
+				newFocus->updateFocusElement(newFocus->available, "$TARGET", country.first);
+				newFocus->updateFocusElement(newFocus->completionReward, "$TARGET", country.first);
+				newFocus->updateFocusElement(newFocus->bypass, "$TARGET", country.first);
+				newFocus->updateFocusElement(newFocus->aiWillDo, "$TARGET", country.first);
+				newFocus->updateFocusElement(newFocus->aiWillDo, "$TAG", theCountry->getTag());
+				newFocus->updateFocusElement(newFocus->aiWillDo, "$REVANCHISM", std::to_string(std::max(static_cast<int>(country.second), 1)));
+				focuses.push_back(newFocus);
+			}
+			else
+			{
+				throw std::runtime_error("Could not load focus declare_war");
+			}
+
+			if (const auto& originalFocus = loadedFocuses.find("cleanup_revanchism"); originalFocus != loadedFocuses.end())
+			{
+				shared_ptr<HoI4Focus> newFocus = originalFocus->second.makeTargetedCopy(theCountry->getTag(), country.first);
+				newFocus->prerequisites.clear();
+				newFocus->prerequisites.push_back("= { focus = declare_war" + theCountry->getTag() + country.first + " }");
+				newFocus->relativePositionId += country.first;
+				newFocus->updateFocusElement(newFocus->available, "$TARGET", country.first);
+				newFocus->updateFocusElement(newFocus->completionReward, "$TARGET", country.first);
+				newFocus->updateFocusElement(newFocus->completionReward, "$REVANCHISM", std::to_string(0.000005*country.second));
+				newFocus->updateFocusElement(newFocus->completionReward, "$STABILITY", std::to_string(0.0000005*country.second));
+				focuses.push_back(newFocus);
+			}
+			else
+			{
+				throw std::runtime_error("Could not load focus cleanup_revanchism");
+			}
+		}
+	}
+
+	return coreHolders;
+}
+
+
+std::set<std::string> HoI4FocusTree::addConquerBranch(
+	std::shared_ptr<HoI4::Country> theCountry,
+	int& numWarsWithNeighbors,
+	const std::set<std::string>& majorIdeologies,
+	std::map<std::string, int> coreHolders
+)
+{
+	std::string tag = theCountry->getTag();
+	std::vector<std::pair<std::string, int>> conquerQue;
+	std::set<std::string> conquerTags;
+
+	int maxConquerValue = 0;
+	for (auto strategy: theCountry->getConquerStrategies())
+	{
+		if (strategy.second.getValue() > maxConquerValue)
+		{
+			maxConquerValue = strategy.second.getValue();
+		}
+	}
+	std::vector<std::pair<std::string, int>> sortedConquerStrategies;
+	for (int i = maxConquerValue; i > 0; i--)
```

Comment:
You can just put them in a vector and then use std::sort to sort it. You can pass in a custom function to make it work how you want:
[](std::pair<std::string, int>a, std::pair<std::string, int>b) {
        return a.second > b.second;   
    });

Reply:
[Use std::sort for sorting conquer strategies](https://github.com/ParadoxGameConverters/Vic2ToHoI4/pull/568/commits/cb728d23792a7be2bf571419261a4aa1f1e35998)